test_regress/driver.py

```diff
@@ -1772,23 +1776,13 @@ def run(
                 (pid, rc) = os.waitpid(pid, 0)
 
         else:
-            with subprocess.Popen(command,
-                                  shell=True,
-                                  bufsize=0,
-                                  stdout=subprocess.PIPE,
-                                  stderr=subprocess.STDOUT) as proc:
-                rawbuf = bytearray(2048)
-
-                while True:
-                    finished = proc.poll()
-                    # Need to check readinto once, even after poll "completes"
-                    got = proc.stdout.readinto(rawbuf)
-                    if got:
-                        data = rawbuf[0:got]
-                        self._run_output(data, logfh, tee)
-                    elif finished is not None:
-                        break
-
+            # Do not redirect output when using an interactive debugger, so it
+            # can have direct access to the user terminal (so terminal control
+            # characters and the like work). That means the log file will be
+            # empty but hopefully that's ok, just re-run the test without the
+            # interactive debugger to confirm a fix.
+            with subprocess.Popen(command, shell=True, bufsize=0) as proc:
+                proc.wait()
```

Comment:
Does the log still get the output?

I'd like this to still pass

```
~/SandBox/homecvs/v4/verilator/test_regress/t/t_flag_quiet_stats3.py --gdb
run
q
```

But maybe I can get over that.

Reply:
> Does the log still get the output?

Unfortunately not.

This is what I meant. I see no easy way to have both (a proper interactive terminal, and the stdout collected the same way as the non-interactive run). Could theoretically tell GDB to write the process output to the logfile, but that sounds like a pain and would still not work for rr for example.

Reply:
I can live without the log.